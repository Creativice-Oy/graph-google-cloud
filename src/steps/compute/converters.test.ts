import {
  createComputeDiskEntity,
  createComputeFirewallEntity,
  createComputeInstanceEntity,
  createComputeInstanceUsesComputeDiskRelationship,
  createComputeNetworkEntity,
  createComputeProjectEntity,
  createComputeSubnetEntity,
  createFirewallRuleMappedRelationship,
  getIpAddressesForComputeInstance,
} from './converters';
import {
  getMockComputeDisk,
  getMockComputeFirewall,
  getMockComputeInstance,
  getMockNetwork,
  getMockSubnet,
  getMockComputeProject,
  getMockBackendBucket,
  getMockBackendService,
  getMockHealthCheck,
  getMockInstanceGroup,
  getMockLoadBalancer,
  getMockTargetHttpProxy,
  getMockTargetHttpsProxy,
  getMockTargetSslProxy,
  getMockSslPolicy,
} from '../../../test/mocks';
import { DEFAULT_INTEGRATION_CONFIG_PROJECT_ID } from '../../../test/config';
import {
  RelationshipClass,
  RelationshipDirection,
} from '@jupiterone/integration-sdk-core';
import { INTERNET } from '@jupiterone/data-model';

describe('#createComputeDiskEntity', () => {
  test('should convert to entity', () => {
    expect(createComputeDiskEntity(getMockComputeDisk())).toMatchSnapshot();
  });

  test('should set active to false when status is not READY', () => {
    expect(
      createComputeDiskEntity(getMockComputeDisk({ status: 'FAILED' })),
    ).toMatchSnapshot();
  });
});

describe('#createComputeInstanceEntity', () => {
  test('should convert to entity', () => {
    expect(
      createComputeInstanceEntity(getMockComputeInstance()),
    ).toMatchSnapshot();
  });

  test('should set active to false when status is not RUNNING', () => {
    expect(
      createComputeInstanceEntity(
        getMockComputeInstance({ status: 'SUSPENDED' }),
      ),
    ).toMatchSnapshot();
  });

  test('should set isOSLoginEnabled to true when enabled-oslogin metadata value is "TRUE"', () => {
    expect(
      createComputeInstanceEntity(
        getMockComputeInstance({
          metadata: {
            items: [
              {
                key: 'enable-oslogin',
                value: 'TRUE',
              },
            ],
            kind: 'compute#metadata',
          },
        }),
      ),
    ).toMatchSnapshot();
  });
});

describe('#createComputeProjectEntity', () => {
  test('should convert to entity', () => {
    expect(
      createComputeProjectEntity(getMockComputeProject()),
    ).toMatchSnapshot();
  });

  test('should set isOSLoginEnabled to true when enabled-oslogin metadata value is "TRUE"', () => {
    expect(
      createComputeProjectEntity(
        getMockComputeProject({
          commonInstanceMetadata: {
            items: [
              {
                key: 'enable-oslogin',
                value: 'TRUE',
              },
            ],
            kind: 'compute#metadata',
          },
        }),
      ),
    ).toMatchSnapshot();
  });
});

describe('#createComputeInstanceUsesComputeDiskRelationship', () => {
  test('should convert to relationship', () => {
    const computeDiskEntity = createComputeDiskEntity(getMockComputeDisk());
    const computeInstance = getMockComputeInstance();
    const computeInstanceEntity = createComputeInstanceEntity(computeInstance);

    expect(
      createComputeInstanceUsesComputeDiskRelationship({
        computeInstanceEntity,
        computeDiskEntity,
        mode: 'READ_WRITE',
        autoDelete: true,
        deviceName: 'persisten-disk-0',
        interface: 'SCSI',
      }),
    ).toMatchSnapshot();
  });
});

describe('#createFirewallRuleMappedRelationship', () => {
  test('should convert to mapped relationship', () => {
    const firewallEntity = createComputeFirewallEntity(
      getMockComputeFirewall(),
    );

    expect(
      createFirewallRuleMappedRelationship({
        _class: RelationshipClass.ALLOWS,
        relationshipDirection: RelationshipDirection.REVERSE,
        targetFilterKeys: [['_key']],
        targetEntity: INTERNET,
        firewallEntity,
        properties: {
          ipRange: '0.0.0.0/0',
          protocol: 'tcp',
          ipProtocol: 'tcp',
          portRange: '443',
          fromPort: 443,
          toPort: 443,
        },
      }),
    ).toMatchSnapshot();
  });
});

describe('#createComputeFirewallEntity', () => {
  test('should convert to entity', () => {
    expect(
      createComputeFirewallEntity(getMockComputeFirewall()),
    ).toMatchSnapshot();
  });
});

describe('#createComputeSubnetEntity', () => {
  test('should convert to entity', () => {
    expect(
      createComputeSubnetEntity(
        getMockSubnet(),
        DEFAULT_INTEGRATION_CONFIG_PROJECT_ID,
      ),
    ).toMatchSnapshot();
  });

  test('should have flowLogsEnabled set to true if it is enabled for subnet', () => {
    expect(
      createComputeSubnetEntity(
        getMockSubnet({
          logConfig: {
            enable: true,
          },
        }),
        DEFAULT_INTEGRATION_CONFIG_PROJECT_ID,
      ),
    ).toMatchSnapshot();
  });
});

describe('#createComputeNetworkEntity', () => {
  test('should convert to entity', () => {
    expect(
      createComputeNetworkEntity(
        getMockNetwork(),
        DEFAULT_INTEGRATION_CONFIG_PROJECT_ID,
      ),
    ).toMatchSnapshot();
  });
});

describe('#getIpAddressesForComputeInstance', () => {
  test('should get IP addresses given a compute instance with network interfaces and accessConfigs', () => {
    expect(
      getIpAddressesForComputeInstance(
        getMockComputeInstance({
          networkInterfaces: [
            {
              network:
                'https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/global/networks/public-compute-app-vpc',
              subnetwork:
                'https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/regions/us-central1/subnetworks/public-compute-app-public-subnet-1',
              networkIP: '10.10.1.2',
              name: 'nic0',
              accessConfigs: [
                {
                  type: 'ONE_TO_ONE_NAT',
                  name: 'external-nat',
                  natIP: '34.71.33.132',
                  networkTier: 'PREMIUM',
                  kind: 'compute#accessConfig',
                },
              ],
              fingerprint: 'ElJkype-dKI=',
              kind: 'compute#networkInterface',
            },
          ],
        }),
      ),
    ).toEqual({
      publicIpAddresses: ['34.71.33.132'],
      privateIpAddresses: ['10.10.1.2'],
    });
  });
});

describe('#createBackendBucketEntity', () => {
  test('should convert to entity', () => {
    expect(getMockBackendBucket()).toMatchSnapshot();
  });
});

describe('#createBackendServiceEntity', () => {
  test('should convert to entity', () => {
    expect(getMockBackendService()).toMatchSnapshot();
  });
});

describe('#createHealthCheckEntity', () => {
  test('should convert to entity', () => {
    expect(getMockHealthCheck()).toMatchSnapshot();
  });
});

describe('#createInstanceGroupEntity', () => {
  test('should convert to entity', () => {
    expect(getMockInstanceGroup()).toMatchSnapshot();
  });
});

describe('#createLoadBalancerEntity', () => {
  test('should convert to entity', () => {
    expect(getMockLoadBalancer()).toMatchSnapshot();
  });
});

describe('#createTargetHttpProxyEntity', () => {
  test('should convert to entity', () => {
    expect(getMockTargetHttpProxy()).toMatchSnapshot();
  });
});

describe('#createTargetHttpsProxyEntity', () => {
  test('should convert to entity', () => {
    expect(getMockTargetHttpsProxy()).toMatchSnapshot();
  });
});

describe('#createTargetSslProxyEntity', () => {
  test('should convert to entity', () => {
    expect(getMockTargetSslProxy()).toMatchSnapshot();
  });
});

describe('#createSslPolicyEntity', () => {
  test('should convert to entity', () => {
    expect(getMockSslPolicy()).toMatchSnapshot();
  });
});
