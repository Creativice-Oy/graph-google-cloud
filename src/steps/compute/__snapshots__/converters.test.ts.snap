// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#createComputeDiskEntity should convert to entity 1`] = `
Object {
  "_class": Array [
    "DataStore",
    "Disk",
  ],
  "_key": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/disks/testvm",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "creationTimestamp": "2020-08-12T08:20:05.930-07:00",
        "guestOsFeatures": Array [
          Object {
            "type": "VIRTIO_SCSI_MULTIQUEUE",
          },
        ],
        "id": "123456789",
        "kind": "compute#disk",
        "labelFingerprint": "42WmSpB8rSM=",
        "lastAttachTimestamp": "2020-08-12T08:20:05.930-07:00",
        "licenseCodes": Array [
          "1000205",
        ],
        "licenses": Array [
          "https://www.googleapis.com/compute/v1/projects/debian-cloud/global/licenses/debian-9-stretch",
        ],
        "name": "testvm",
        "physicalBlockSizeBytes": "4096",
        "selfLink": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/disks/testvm",
        "sizeGb": "10",
        "sourceImage": "https://www.googleapis.com/compute/v1/projects/debian-cloud/global/images/debian-9-stretch-v20200805",
        "sourceImageId": "6709658075886210235",
        "status": "READY",
        "tags": undefined,
        "type": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/diskTypes/pd-standard",
        "users": Array [
          "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/instances/testvm",
        ],
        "zone": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a",
      },
    },
  ],
  "_type": "google_compute_disk",
  "active": true,
  "classification": null,
  "createdOn": 1597245605930,
  "description": undefined,
  "displayName": "testvm",
  "encrypted": true,
  "guestOsFeatures": Array [
    "VIRTIO_SCSI_MULTIQUEUE",
  ],
  "id": "123456789",
  "isCustomerSuppliedKeysEncrypted": false,
  "kind": "compute#disk",
  "labelFingerprint": "42WmSpB8rSM=",
  "lastAttachTimestamp": 1597245605930,
  "licenseCodes": Array [
    "1000205",
  ],
  "licenses": Array [
    "https://www.googleapis.com/compute/v1/projects/debian-cloud/global/licenses/debian-9-stretch",
  ],
  "name": "testvm",
  "physicalBlockSizeBytes": "4096",
  "sizeGB": "10",
  "sourceImage": "https://www.googleapis.com/compute/v1/projects/debian-cloud/global/images/debian-9-stretch-v20200805",
  "sourceImageId": "6709658075886210235",
  "status": "READY",
  "type": "pd-standard",
  "zone": "us-central1-a",
}
`;

exports[`#createComputeDiskEntity should set active to false when status is not READY 1`] = `
Object {
  "_class": Array [
    "DataStore",
    "Disk",
  ],
  "_key": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/disks/testvm",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "creationTimestamp": "2020-08-12T08:20:05.930-07:00",
        "guestOsFeatures": Array [
          Object {
            "type": "VIRTIO_SCSI_MULTIQUEUE",
          },
        ],
        "id": "123456789",
        "kind": "compute#disk",
        "labelFingerprint": "42WmSpB8rSM=",
        "lastAttachTimestamp": "2020-08-12T08:20:05.930-07:00",
        "licenseCodes": Array [
          "1000205",
        ],
        "licenses": Array [
          "https://www.googleapis.com/compute/v1/projects/debian-cloud/global/licenses/debian-9-stretch",
        ],
        "name": "testvm",
        "physicalBlockSizeBytes": "4096",
        "selfLink": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/disks/testvm",
        "sizeGb": "10",
        "sourceImage": "https://www.googleapis.com/compute/v1/projects/debian-cloud/global/images/debian-9-stretch-v20200805",
        "sourceImageId": "6709658075886210235",
        "status": "FAILED",
        "tags": undefined,
        "type": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/diskTypes/pd-standard",
        "users": Array [
          "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/instances/testvm",
        ],
        "zone": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a",
      },
    },
  ],
  "_type": "google_compute_disk",
  "active": false,
  "classification": null,
  "createdOn": 1597245605930,
  "description": undefined,
  "displayName": "testvm",
  "encrypted": true,
  "guestOsFeatures": Array [
    "VIRTIO_SCSI_MULTIQUEUE",
  ],
  "id": "123456789",
  "isCustomerSuppliedKeysEncrypted": false,
  "kind": "compute#disk",
  "labelFingerprint": "42WmSpB8rSM=",
  "lastAttachTimestamp": 1597245605930,
  "licenseCodes": Array [
    "1000205",
  ],
  "licenses": Array [
    "https://www.googleapis.com/compute/v1/projects/debian-cloud/global/licenses/debian-9-stretch",
  ],
  "name": "testvm",
  "physicalBlockSizeBytes": "4096",
  "sizeGB": "10",
  "sourceImage": "https://www.googleapis.com/compute/v1/projects/debian-cloud/global/images/debian-9-stretch-v20200805",
  "sourceImageId": "6709658075886210235",
  "status": "FAILED",
  "type": "pd-standard",
  "zone": "us-central1-a",
}
`;

exports[`#createComputeFirewallEntity should convert to entity 1`] = `
Object {
  "_class": Array [
    "Firewall",
  ],
  "_key": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/global/firewalls/public-compute-app-fw-allow-https",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "allowed": Array [
          Object {
            "IPProtocol": "tcp",
            "ports": Array [
              "443",
            ],
          },
        ],
        "creationTimestamp": "2020-09-23T07:47:24.809-07:00",
        "description": "",
        "direction": "INGRESS",
        "disabled": false,
        "id": "1234567890",
        "kind": "compute#firewall",
        "logConfig": Object {
          "enable": false,
        },
        "name": "public-compute-app-fw-allow-https",
        "network": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/global/networks/public-compute-app-vpc",
        "priority": 1000,
        "selfLink": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/global/firewalls/public-compute-app-fw-allow-https",
        "sourceRanges": Array [
          "0.0.0.0/0",
        ],
        "targetTags": Array [
          "https",
        ],
      },
    },
  ],
  "_type": "google_compute_firewall",
  "allowed": "[{\\"IPProtocol\\":\\"tcp\\",\\"ports\\":[\\"443\\"]}]",
  "category": Array [
    "network",
  ],
  "createdOn": 1600872444809,
  "denied": undefined,
  "description": "",
  "destinationRanges": undefined,
  "disabled": false,
  "displayName": "public-compute-app-fw-allow-https",
  "egress": false,
  "id": "1234567890",
  "ingress": true,
  "internal": true,
  "logConfigEnabled": false,
  "logConfigMetadata": undefined,
  "name": "public-compute-app-fw-allow-https",
  "network": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/global/networks/public-compute-app-vpc",
  "priority": 1000,
  "public": false,
  "sourceRanges": Array [
    "0.0.0.0/0",
  ],
  "sourceServiceAccounts": undefined,
  "sourceTags": undefined,
  "targetServiceAccounts": undefined,
  "targetTags": Array [
    "https",
  ],
}
`;

exports[`#createComputeInstanceEntity should convert to entity 1`] = `
Object {
  "_class": Array [
    "Host",
  ],
  "_key": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/instances/testvm",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "canIpForward": false,
        "cpuPlatform": "Intel Haswell",
        "creationTimestamp": "2020-08-12T08:20:05.921-07:00",
        "deletionProtection": false,
        "disks": Array [
          Object {
            "autoDelete": true,
            "boot": true,
            "deviceName": "persistent-disk-0",
            "diskSizeGb": "10",
            "guestOsFeatures": Array [
              Object {
                "type": "VIRTIO_SCSI_MULTIQUEUE",
              },
            ],
            "index": 0,
            "interface": "SCSI",
            "kind": "compute#attachedDisk",
            "licenses": Array [
              "https://www.googleapis.com/compute/v1/projects/debian-cloud/global/licenses/debian-9-stretch",
            ],
            "mode": "READ_WRITE",
            "source": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/disks/testvm",
            "type": "PERSISTENT",
          },
          Object {
            "autoDelete": true,
            "boot": false,
            "deviceName": "local-ssd-0",
            "diskSizeGb": "375",
            "index": 1,
            "interface": "SCSI",
            "kind": "compute#attachedDisk",
            "mode": "READ_WRITE",
            "type": "SCRATCH",
          },
        ],
        "fingerprint": "4gmEF7tlfn8=",
        "id": "2248191918776890954",
        "kind": "compute#instance",
        "labelFingerprint": "42WmSpB8rSM=",
        "machineType": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/machineTypes/n1-standard-1",
        "metadata": Object {
          "fingerprint": "9yLGkjCLBTs=",
          "items": Array [
            Object {
              "key": "foo",
              "value": "bar",
            },
            Object {
              "key": "startup-script",
              "value": "echo Hello",
            },
          ],
          "kind": "compute#metadata",
        },
        "name": "testvm",
        "networkInterfaces": Array [
          Object {
            "fingerprint": "RpvQNHzHRp8=",
            "kind": "compute#networkInterface",
            "name": "nic0",
            "network": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/global/networks/default",
            "networkIP": "10.128.0.2",
            "subnetwork": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/regions/us-central1/subnetworks/default",
          },
        ],
        "scheduling": Object {
          "automaticRestart": true,
          "onHostMaintenance": "MIGRATE",
          "preemptible": false,
        },
        "selfLink": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/instances/testvm",
        "serviceAccounts": Array [
          Object {
            "email": "j1-gc-integration-dev-sa-tf@j1-gc-integration-dev.iam.gserviceaccount.com",
            "scopes": Array [
              "https://www.googleapis.com/auth/compute.readonly",
              "https://www.googleapis.com/auth/devstorage.read_only",
              "https://www.googleapis.com/auth/userinfo.email",
            ],
          },
        ],
        "startRestricted": false,
        "status": "RUNNING",
        "tags": Array [
          Object {
            "testvmtag1": true,
          },
          Object {
            "testvmtag2": true,
          },
        ],
        "zone": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a",
      },
    },
  ],
  "_type": "google_compute_instance",
  "active": true,
  "blockProjectSSHKeys": undefined,
  "canIpForward": false,
  "cpuPlatform": "Intel Haswell",
  "createdOn": 1597245605921,
  "deletionProtection": false,
  "displayName": "testvm",
  "fingerprint": "4gmEF7tlfn8=",
  "id": "2248191918776890954",
  "isSerialPortEnabled": undefined,
  "isShieldedVM": undefined,
  "kind": "compute#instance",
  "labelFingerprint": "42WmSpB8rSM=",
  "machineType": "n1-standard-1",
  "name": "testvm",
  "privateIpAddress": Array [
    "10.128.0.2",
  ],
  "publicIpAddress": Array [],
  "startRestricted": false,
  "status": "RUNNING",
  "usesDefaultServiceAccount": false,
  "usesFullAccessDefaultServiceAccount": false,
  "zone": "us-central1-a",
}
`;

exports[`#createComputeInstanceEntity should set active to false when status is not RUNNING 1`] = `
Object {
  "_class": Array [
    "Host",
  ],
  "_key": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/instances/testvm",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "canIpForward": false,
        "cpuPlatform": "Intel Haswell",
        "creationTimestamp": "2020-08-12T08:20:05.921-07:00",
        "deletionProtection": false,
        "disks": Array [
          Object {
            "autoDelete": true,
            "boot": true,
            "deviceName": "persistent-disk-0",
            "diskSizeGb": "10",
            "guestOsFeatures": Array [
              Object {
                "type": "VIRTIO_SCSI_MULTIQUEUE",
              },
            ],
            "index": 0,
            "interface": "SCSI",
            "kind": "compute#attachedDisk",
            "licenses": Array [
              "https://www.googleapis.com/compute/v1/projects/debian-cloud/global/licenses/debian-9-stretch",
            ],
            "mode": "READ_WRITE",
            "source": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/disks/testvm",
            "type": "PERSISTENT",
          },
          Object {
            "autoDelete": true,
            "boot": false,
            "deviceName": "local-ssd-0",
            "diskSizeGb": "375",
            "index": 1,
            "interface": "SCSI",
            "kind": "compute#attachedDisk",
            "mode": "READ_WRITE",
            "type": "SCRATCH",
          },
        ],
        "fingerprint": "4gmEF7tlfn8=",
        "id": "2248191918776890954",
        "kind": "compute#instance",
        "labelFingerprint": "42WmSpB8rSM=",
        "machineType": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/machineTypes/n1-standard-1",
        "metadata": Object {
          "fingerprint": "9yLGkjCLBTs=",
          "items": Array [
            Object {
              "key": "foo",
              "value": "bar",
            },
            Object {
              "key": "startup-script",
              "value": "echo Hello",
            },
          ],
          "kind": "compute#metadata",
        },
        "name": "testvm",
        "networkInterfaces": Array [
          Object {
            "fingerprint": "RpvQNHzHRp8=",
            "kind": "compute#networkInterface",
            "name": "nic0",
            "network": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/global/networks/default",
            "networkIP": "10.128.0.2",
            "subnetwork": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/regions/us-central1/subnetworks/default",
          },
        ],
        "scheduling": Object {
          "automaticRestart": true,
          "onHostMaintenance": "MIGRATE",
          "preemptible": false,
        },
        "selfLink": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/instances/testvm",
        "serviceAccounts": Array [
          Object {
            "email": "j1-gc-integration-dev-sa-tf@j1-gc-integration-dev.iam.gserviceaccount.com",
            "scopes": Array [
              "https://www.googleapis.com/auth/compute.readonly",
              "https://www.googleapis.com/auth/devstorage.read_only",
              "https://www.googleapis.com/auth/userinfo.email",
            ],
          },
        ],
        "startRestricted": false,
        "status": "SUSPENDED",
        "tags": Array [
          Object {
            "testvmtag1": true,
          },
          Object {
            "testvmtag2": true,
          },
        ],
        "zone": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a",
      },
    },
  ],
  "_type": "google_compute_instance",
  "active": false,
  "blockProjectSSHKeys": undefined,
  "canIpForward": false,
  "cpuPlatform": "Intel Haswell",
  "createdOn": 1597245605921,
  "deletionProtection": false,
  "displayName": "testvm",
  "fingerprint": "4gmEF7tlfn8=",
  "id": "2248191918776890954",
  "isSerialPortEnabled": undefined,
  "isShieldedVM": undefined,
  "kind": "compute#instance",
  "labelFingerprint": "42WmSpB8rSM=",
  "machineType": "n1-standard-1",
  "name": "testvm",
  "privateIpAddress": Array [
    "10.128.0.2",
  ],
  "publicIpAddress": Array [],
  "startRestricted": false,
  "status": "SUSPENDED",
  "usesDefaultServiceAccount": false,
  "usesFullAccessDefaultServiceAccount": false,
  "zone": "us-central1-a",
}
`;

exports[`#createComputeInstanceUsesComputeDiskRelationship should convert to relationship 1`] = `
Object {
  "_class": "USES",
  "_fromEntityKey": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/instances/testvm",
  "_key": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/instances/testvm|uses|https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/disks/testvm",
  "_toEntityKey": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/zones/us-central1-a/disks/testvm",
  "_type": "google_compute_instance_uses_disk",
  "autoDelete": true,
  "deviceName": "persisten-disk-0",
  "displayName": "USES",
  "interface": "SCSI",
  "mode": "READ_WRITE",
}
`;

exports[`#createComputeNetworkEntity should convert to entity 1`] = `
Object {
  "CIDR": null,
  "IPv4Range": undefined,
  "_class": Array [
    "Network",
  ],
  "_key": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/global/networks/public-compute-app-vpc",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "autoCreateSubnetworks": false,
        "creationTimestamp": "2020-09-23T07:47:13.320-07:00",
        "id": "4319333979087364382",
        "kind": "compute#network",
        "name": "public-compute-app-vpc",
        "routingConfig": Object {
          "routingMode": "GLOBAL",
        },
        "selfLink": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/global/networks/public-compute-app-vpc",
        "subnetworks": Array [
          "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/regions/us-central1/subnetworks/public-compute-app-public-subnet-1",
        ],
      },
    },
  ],
  "_type": "google_compute_network",
  "autoCreateSubnetworks": false,
  "createdOn": 1600872433320,
  "description": undefined,
  "displayName": "public-compute-app-vpc",
  "gatewayIPv4": undefined,
  "id": "4319333979087364382",
  "internal": true,
  "name": "public-compute-app-vpc",
  "public": false,
  "routingMode": "GLOBAL",
  "subnetworks": Array [
    "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/regions/us-central1/subnetworks/public-compute-app-public-subnet-1",
  ],
  "webLink": "https://console.cloud.google.com/networking/networks/details/public-compute-app-vpc?project=j1-gc-integration-dev",
}
`;

exports[`#createComputeSubnetEntity should convert to entity 1`] = `
Object {
  "CIDR": "10.164.0.0/20",
  "_class": Array [
    "Network",
  ],
  "_key": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/regions/europe-west4/subnetworks/default",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "creationTimestamp": "2020-08-07T10:20:16.033-07:00",
        "fingerprint": "whfZBEBCda4=",
        "gatewayAddress": "10.164.0.1",
        "id": "1234567890",
        "ipCidrRange": "10.164.0.0/20",
        "kind": "compute#subnetwork",
        "name": "default",
        "network": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/global/networks/default",
        "privateIpGoogleAccess": false,
        "privateIpv6GoogleAccess": "DISABLE_GOOGLE_ACCESS",
        "purpose": "PRIVATE",
        "region": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/regions/europe-west4",
        "selfLink": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/regions/europe-west4/subnetworks/default",
      },
    },
  ],
  "_type": "google_compute_subnetwork",
  "createdOn": 1596820816033,
  "description": undefined,
  "displayName": "default",
  "flowLogsEnabled": false,
  "gatewayAddress": "10.164.0.1",
  "id": "1234567890",
  "internal": true,
  "name": "default",
  "privateIpGoogleAccess": false,
  "public": false,
  "purpose": "PRIVATE",
  "webLink": "https://console.cloud.google.com/networking/subnetworks/details/europe-west4/default?project=j1-gc-integration-dev",
}
`;

exports[`#createComputeSubnetEntity should have flowLogsEnabled set to true if it is enabled for subnet 1`] = `
Object {
  "CIDR": "10.164.0.0/20",
  "_class": Array [
    "Network",
  ],
  "_key": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/regions/europe-west4/subnetworks/default",
  "_rawData": Array [
    Object {
      "name": "default",
      "rawData": Object {
        "creationTimestamp": "2020-08-07T10:20:16.033-07:00",
        "fingerprint": "whfZBEBCda4=",
        "gatewayAddress": "10.164.0.1",
        "id": "1234567890",
        "ipCidrRange": "10.164.0.0/20",
        "kind": "compute#subnetwork",
        "logConfig": Object {
          "enable": true,
        },
        "name": "default",
        "network": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/global/networks/default",
        "privateIpGoogleAccess": false,
        "privateIpv6GoogleAccess": "DISABLE_GOOGLE_ACCESS",
        "purpose": "PRIVATE",
        "region": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/regions/europe-west4",
        "selfLink": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/regions/europe-west4/subnetworks/default",
      },
    },
  ],
  "_type": "google_compute_subnetwork",
  "createdOn": 1596820816033,
  "description": undefined,
  "displayName": "default",
  "flowLogsEnabled": true,
  "gatewayAddress": "10.164.0.1",
  "id": "1234567890",
  "internal": true,
  "name": "default",
  "privateIpGoogleAccess": false,
  "public": false,
  "purpose": "PRIVATE",
  "webLink": "https://console.cloud.google.com/networking/subnetworks/details/europe-west4/default?project=j1-gc-integration-dev",
}
`;

exports[`#createFirewallRuleMappedRelationship should convert to mapped relationship 1`] = `
Object {
  "_class": "ALLOWS",
  "_key": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/global/firewalls/public-compute-app-fw-allow-https:tcp:0.0.0.0/0:443",
  "_mapping": Object {
    "relationshipDirection": "REVERSE",
    "skipTargetCreation": true,
    "sourceEntityKey": "https://www.googleapis.com/compute/v1/projects/j1-gc-integration-dev/global/firewalls/public-compute-app-fw-allow-https",
    "targetEntity": Object {
      "CIDR": "0.0.0.0/0",
      "CIDRv6": "::/0",
      "_class": Array [
        "Internet",
        "Network",
      ],
      "_key": "global:internet",
      "_type": "internet",
      "displayName": "Internet",
      "public": true,
    },
    "targetFilterKeys": Array [
      Array [
        "_key",
      ],
    ],
  },
  "_type": "google_cloud_firewall_rule",
  "displayName": "ALLOWS",
  "fromPort": 443,
  "ipProtocol": "tcp",
  "ipRange": "0.0.0.0/0",
  "portRange": "443",
  "protocol": "tcp",
  "toPort": 443,
}
`;
